src/cascadia/TerminalSettingsModel/Command.cpp

```diff
@@ -741,4 +741,57 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
 
         return winrt::single_threaded_vector<Model::Command>(std::move(result));
     }
+
+    void Command::LogSettingChanges(std::set<std::string>& changes)

```

Comment:
```json
{ "command": { "action": "copy", "singleLine": false }, "description": "My Cool Copy" }
```

This results in 3 entries to `changes`: one for the description and 2 for each of the args. Should we just do 1 entry per non-iterating command block? It would require some concatenation of the args/description but I'm not sure why we're adding so many changes for effectively "one" setting

Reply:
Nah, it results in 2 entries `action.copy.singleLine` and `action.description`.
- `action.description` just makes sense to track once, regardless of which action is selected. My thinking is that we don't really care about _what_ action the description was added to. Just that people are using `description` and are aware that feature exists.
- `action.copy.singleLine` tracks that people are using the `singleLine` arg on `copy`. If someone added the `copyFormatting` arg too, we'd get two entries: `action.copy.singleLine` and `action.copy.copyFormatting`. We _could_ combine them into one event, but I'm trying to have the same philosophy throughout of tracking if the user is aware that a feature/setting exists, not really the combination of said features (i.e. having `action.copy.singleLine.copyFormatting` would report that users are using _both_ at the same time).

Reply:
(oops forgot to add the second arg in my action example yeah)

> if the user is aware that a feature/setting exists, not really the combination of said features

Makes sense!